pennylane/measurements/measurements.py

```diff
@@ -356,10 +359,17 @@ def eigvals(self):
             array: eigvals representation
         """
         if self.mv is not None:
+            if getattr(self.mv, "name", None) == "MeasurementValue":
+                # Indexing a MeasurementValue gives the output of the processing function
+                # for the binary number corresponding to the index.
+                return qml.math.asarray([self.mv[i] for i in range(2 ** len(self.wires))])
             return qml.math.arange(0, 2 ** len(self.wires), 1)
 
         if self.obs is not None:
-            return self.obs.eigvals()
+            try:
+                return qml.eigvals(self.obs)
+            except DecompositionUndefinedError as e:
+                raise EigvalsUndefinedError from e
```

Comment:
Why this change?

Reply:
Basically wanted to replace the use of `qml.eigvals` and `obs.eigvals` with `mp.eigvals` as much as possible without impacting which error is raised. `CountsMP.process_samples` for example used `qml.eigvals(obs)`, which wasn't compatible with `MeasurementValue`s, so I replaced it with `self.eigvals`, but that caused test failures because `MeasurementProcess.eigvals` uses `obs.eigvals`, not `qml.eigvals(obs)`.